core/src/main/java/io/questdb/cairo/GenericRecordMetadata.java

```diff
@@ -31,7 +31,9 @@ public class GenericRecordMetadata extends AbstractRecordMetadata {
 
     public static void copyColumns(RecordMetadata from, GenericRecordMetadata to) {
         for (int i = 0, n = from.getColumnCount(); i < n; i++) {
-            to.add(from.getColumnMetadata(i));
+            if (from.hasColumn(i)) {
```

Comment:
I was curious why deleted columns be in the metadata in the first place. Even if delete columns are there, nothing tells the downstream code not to do this check. 

Sure enough, removing this condition does not fail any test on Windows. Running on Mac

Reply:
> Sure enough, removing this condition does not fail any test on Windows. Running on Mac

I'll add a test to `IODispatcherTest` to fix that.

Reply:
Added the test in 0d5022612cef60d9846ae5cb293cf0635e8b8c45

Reply:
outdated